module-services/service-desktop/Sync.cpp

```diff
@@ -0,0 +1,205 @@
+    std::error_code errorCode;
+
+    for (auto &direntry : std::filesystem::directory_iterator(path)) {
+        if (!isValidDirentry(direntry)) {
+            continue;
+        }
+
+        LOG_INFO("Archiving file ...");
+        auto *file = std::fopen(direntry.path().string().c_str(), "r");
+
+        if (file == nullptr) {
+            LOG_ERROR("Archiving file failed, cannot open file, quitting...");
+            mtar_close(&tarFile);
+            Sync::RemoveSyncDir(path);
+            return false;
+        }
+
+        LOG_DEBUG("Writting tar header ...");
+
+        if (mtar_write_file_header(&tarFile,
+                                   direntry.path().filename().c_str(),
+                                   static_cast<unsigned>(std::filesystem::file_size(direntry))) != MTAR_ESUCCESS) {
+            LOG_ERROR("Writing tar header failed");
+            std::fclose(file);
+            mtar_close(&tarFile);
+            Sync::RemoveSyncDir(path);
+            return false;
+        }
+
+        uintmax_t filesize = std::filesystem::file_size(direntry.path(), errorCode);
+        if (errorCode) {
+            LOG_ERROR("Failed to get size for file %s, error: %d", direntry.path().c_str(), errorCode.value());
+            Sync::RemoveSyncDir(path);
+            return false;
+        }
+        uint32_t loopcount = (filesize / purefs::buffer::tar_buf) + 1u;
+        uint32_t readsize;
+
+        for (uint32_t i = 0u; i < loopcount; i++) {
+            if (i + 1u == loopcount) {
+                readsize = filesize % purefs::buffer::tar_buf;
+            }
+            else {
+                readsize = purefs::buffer::tar_buf;
+            }
+
+            LOG_DEBUG("Reading file ...");
+
+            if (std::fread(buffer.get(), 1, readsize, file) != readsize) {
+                LOG_ERROR("Reading file failed, quitting...");
+                std::fclose(file);
+                mtar_close(&tarFile);
+                Sync::RemoveSyncDir(path);
+                return false;
+            }
+
+            LOG_DEBUG("Writting into sync package...");
+            if (mtar_write_data(&tarFile, buffer.get(), readsize) != MTAR_ESUCCESS) {
+                LOG_ERROR("Writting into sync package failed, quitting...");
+                std::fclose(file);
+                mtar_close(&tarFile);
+                Sync::RemoveSyncDir(path);
+                return false;
+            }
+        }
+
+        LOG_INFO("Closing file...");
+        if (std::fclose(file) != 0) {
+            LOG_ERROR("Closing file failed, quitting...");
+            mtar_close(&tarFile);
+            Sync::RemoveSyncDir(path);
+            return false;
+        }
+
+        LOG_INFO("Deleting file ...");
+
+        if (!std::filesystem::remove(direntry.path(), errorCode)) {
+            LOG_ERROR("Deleting file failed, error: %d, quitting...", errorCode.value());
+            mtar_close(&tarFile);
+            Sync::RemoveSyncDir(path);
+            return false;
```

Comment:
Why don't delete all the files at the end of the operation in a bulk?
Is this really a good reason to return false?

Reply:
Now all files are deleted at the end of the operation.